sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -322,129 +331,150 @@ static int get_operation_status()
+
+  // Operation is complete
+  // Successful assignment - print out the assigned hostname and device id
+  if (operation_status == AZ_IOT_PROVISIONING_STATUS_ASSIGNED)
+  {
+    printf("SUCCESS - Device provisioned:\n");
+    print_az_span("\tHub Hostname: ", response.registration_result.assigned_hub_hostname);
+    print_az_span("\tDevice Id: ", response.registration_result.device_id);
+  }
+  else // Unsuccesful assignment (unassigned, failed or disabled states)
+  {
+    printf("ERROR - Device Provisioning failed:\n");
+    print_az_span("\tRegistration state: ", response.operation_status);
+    printf("\tLast operation status: %d\n", response.status);
+    print_az_span("\tOperation ID: ", response.operation_id);
+    printf("\tError code: %u\n", response.registration_result.extended_error_code);
+    print_az_span("\tError message: ", response.registration_result.error_message);
+    print_az_span("\tError timestamp: ", response.registration_result.error_timestamp);
+    print_az_span("\tError tracking ID: ", response.registration_result.error_tracking_id);
+
+    if (response.retry_after_seconds > 0)
     {
-      // Successful assignment - print out the assigned hostname and device id
-      if (operation_status == AZ_IOT_PROVISIONING_STATUS_ASSIGNED)
-      {
-        printf("SUCCESS - Device provisioned:\n");
-        printf("\tHub Hostname: ");
-        print_az_span(response.registration_result.assigned_hub_hostname);
-        printf("\tDevice Id: ");
-        print_az_span(response.registration_result.device_id);
-      }
-      else // unassigned, failed or disabled states
-      {
-        printf("ERROR - Device Provisioning failed:\n");
-        printf("\tRegistration state: ");
-        print_az_span(response.operation_status);
-        printf("\tLast operation status: %d\n", response.status);
-        printf("\tOperation ID: ");
-        print_az_span(response.operation_id);
-        printf("\tError code: %u\n", response.registration_result.extended_error_code);
-        printf("\tError message: ");
-        print_az_span(response.registration_result.error_message);
-        printf("\tError timestamp: ");
-        print_az_span(response.registration_result.error_timestamp);
-        printf("\tError tracking ID: ");
-        print_az_span(response.registration_result.error_tracking_id);
-        if (response.retry_after_seconds > 0)
-        {
-          printf("\tRetry-after: %u seconds.", response.retry_after_seconds);
-        }
-      }
+      printf("\tRetry-after: %u seconds.", response.retry_after_seconds);
     }
+  }
+
+  MQTTClient_freeMessage(&message);
+  MQTTClient_free(topic);
+  return rc;
+}
+
+static az_result parse_operation_message(
+    char* topic,
+    int topic_len,
+    MQTTClient_message const* message,
+    az_iot_provisioning_client_register_response* response,
+    az_iot_provisioning_client_operation_status* operation_status)
+{
+  int rc;
+
+  printf("Topic: %s\n", topic);
 
```

Comment:
can this be printed at line 430 using print_span function?

Reply:
yes. nice catch.